tools/travis-install-protoc.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# ensure that the specified version of protoc is installed in
+# /tmp/proto$PROTO_VERSION/bin/protoc, which maybe cached
+
+# bash tools/travis-install-protoc.sh 3.0.0-beta-3.1
+#   
+
+
+# make bash more robust.
+set -beEux -o pipefail
```

Comment:
This is some black magic here...

Did you get this script from somewhere?  If so, put a link to the source in the header comment.  If not, add a comment here about what this line is doing